gtsfm/data_association/point3d_initializer.py

```diff
@@ -210,9 +210,9 @@ def triangulate(self, track_2d: SfmTrack2d) -> Tuple[Optional[SfmTrack], Optiona
             self.track_camera_dict, triangulated_pt, inlier_track.measurements
         )
 
-        # Check that all the measurements have reprojection error < threshold.
-        # TODO(johnwlambert): compare with approach where we only throw away the outlier measurements.
-        if not np.all(reproj_errors < self.reproj_error_thresh):
+        # Filter measurements by the reprojection error.
+        inlier_track = inlier_track.select_subset(np.where(reproj_errors.flatten() < self.reproj_error_thresh)[0])
+        if inlier_track.number_measurements() < 2:
```

Comment:
Should this not be the min track length parameter? Why 2?

Reply:
Min track length is checked outside of the point3d_initializer. See [#260](https://github.com/borglab/gtsfm/pull/260).

Reply:
A check for min track length can be added to the point3d_initializer, but I think that's more in the scope of the other PR.

Reply:
Yeah I think we need to pass that parameter here.